src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -477,15 +477,26 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
             // XXX: don't know what URI to choose...
             vscode.Uri.parse(templateInvoke.templatePath!)
 
-        let awsCredentials: Credentials | undefined
-
-        if (config.aws?.credentials) {
-            const credentials = fromString(config.aws.credentials)
+        let awsCredentials = await this.ctx.awsContext.getCredentials()
+        if (!awsCredentials && !config.aws?.credentials) {
+            getLogger().warn('SAM debug: missing AWS credentials (Toolkit is not connected)')
+        } else if (config.aws?.credentials) {
+            // "aws.credentials" defined in the launch-config takes precedence
+            // over Toolkit's current active credentials.
+            let fromStore: Credentials | undefined
             try {
-                awsCredentials = await getCredentialsFromStore(credentials, this.ctx.credentialsStore)
-            } catch (err) {
-                getLogger().error(err as Error)
-                notifyUserInvalidCredentials(credentials)
+                const credentialsId = fromString(config.aws.credentials)
+                fromStore = await getCredentialsFromStore(credentialsId, this.ctx.credentialsStore)
+            } catch {
+                getLogger().error(`SAM debug: fromString('${config.aws.credentials}') failed`)
```

Comment:
nit: a subsequent call in `getCredentialsFromStore` could still throw which would then get swallowed here, making it seem like `fromString` failed when it actually didn't

Reply:
afaict, `getCredentialsFromStore` can't throw (after the changes in this PR). it has a deep call-tree though... did I miss something?

Reply:
> did I miss something?

Nope. The current implementations won't throw. But there's little stopping someone from implementing `CredentialsProvider` and letting an error slip through `isAvailable`. Not really a big deal though because a failing provider would be caught by other code paths.